Align integer field ranges with CrateDB limits

vidhiii1711 opened the pull request:
This PR verifies and updates integer field range definitions to ensure they align with CrateDB’s actual storage limits, rather than relying solely on Django’s generic field documentation.

While reviewing the existing hardcoded ranges, I cross-checked each integer field against both Django field definitions and CrateDB data type specifications. During this verification, I found mismatches where Django’s documented ranges do not fully match CrateDB’s backend constraints. Therefore, I updated the following fields:

BigIntegerField

PositiveBigIntegerField

PositiveIntegerField

BigAutoField

Each change is based on CrateDB’s supported integer types.

Testing note:

I did not run the integration tests locally, as they require a running CrateDB instance. I’m still getting familiar with the database setup, and this PR focuses on correcting static range definitions only, without modifying runtime database logic.

References:

-> Django integer field documentation:
https://docs.djangoproject.com/en/stable/ref/models/fields/#integerfield

-> CrateDB data type limits:
https://cratedb.com/docs/crate/reference/en/latest/general/ddl/data-types.html

This change is intended to make the backend behavior more accurate and prevent subtle edge-case issues caused by incorrect or assumed integer boundaries.****

surister commented on cratedb_django/operations.py:
```diff
@@ -6,13 +6,13 @@ class DatabaseOperations(BaseDatabaseOperations):
         "IntegerField": (-2147483648, 2147483647),
-        "BigIntegerField": (-9223372036854775808, 9223372036854775807),
-        "PositiveBigIntegerField": (0, 9223372036854775807),
+        "BigIntegerField": (-9223372036854775808, 9223372036854775806),   # 807->806
```
I think it's best overall to remove these comments as they do not offer any new information.

vidhiii1711 commented:
Thanks for the clarification. I’ve fixed PositiveIntegerField to allow only positive values as per Django semantics and removed the unnecessary inline comments. Please let me know if anything else needs adjustment.

vidhiii1711 commented:
Sure, I’ve removed the trailing spaces and the unnecessary `+` signs—thanks for pointing it out.

surister reviewed (changes requested):
Sorry I didn't mean the space after the comma, those were alright. I meant the spaces at the end of the lines:

<img width="836" height="446" alt="Image" src="https://github.com/user-attachments/assets/322decda-0014-4879-ba40-ee167f44f31e" />

This is ok: `"SmallIntegerField": (-32768, 32767),`
This is *not* ok:  `"SmallIntegerField": (-32768, 32767),  `

vidhiii1711 commented:
All requested changes have been addressed. Please let me know if anything else is needed.

surister reviewed (approved):
LGTM thank you

vidhiii1711 commented:
Thanks to you too for your guidance!
